From 711d4ba65ca26242fec882d91606bb68cd9cdf77 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Sun, 16 Feb 2020 21:55:39 +0100
Subject: [PATCH] appDisplay: Properly hide overlay scrollbar in folders

Use the scroll-view policy ST_POLICY_EXTERNAL to hide the scrollbar
instead of setting its css properties to hidden, where it can still be
clicked but isn't visible.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/1011
---
 data/theme/gnome-shell-sass/widgets/_app-grid.scss | 4 ----
 js/ui/appDisplay.js                                | 2 +-
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/data/theme/gnome-shell-sass/widgets/_app-grid.scss b/data/theme/gnome-shell-sass/widgets/_app-grid.scss
index 0209108024..b23bea981b 100644
--- a/data/theme/gnome-shell-sass/widgets/_app-grid.scss
+++ b/data/theme/gnome-shell-sass/widgets/_app-grid.scss
@@ -73,10 +73,6 @@ $app_grid_fg_color: #fff;
       & > StIcon { icon-size: 16px }
     }
   }
-
-  & StButton#vhandle,
-  & StButton#vhandle:hover,
-  & StButton#vhandle:active { background-color: transparent; }
 }
 .app-folder-dialog-container {
   padding: 12px;
diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index 17d69ced9c..4241bd7fb9 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -1308,7 +1308,7 @@ class FolderView extends BaseAppView {
             x_expand: true,
             y_expand: true,
         });
-        this._scrollView.set_policy(St.PolicyType.NEVER, St.PolicyType.AUTOMATIC);
+        this._scrollView.set_policy(St.PolicyType.NEVER, St.PolicyType.EXTERNAL);
         this.add_actor(this._scrollView);
 
         let scrollableContainer = new St.BoxLayout({
-- 
2.24.1

